local_planner/src/nodes/local_planner_node.cpp

```diff
@@ -153,9 +161,21 @@ size_t LocalPlannerNode::numReceivedClouds() {
   return num_received_clouds;
 }
 
+size_t LocalPlannerNode::numTransformedClouds() {
+  size_t num_transformed_clouds = 0;
+  for (size_t i = 0; i < cameras_.size(); i++) {
+    std::unique_lock<std::mutex> lk(*(cameras_[i].trans_ready_mutex_));
+    cameras_[i].trans_ready_cv_->wait_for(
+        lk, std::chrono::milliseconds(30),
```

Comment:
Having this time hardcoded may cause problems on some systems. Is there some way we can do this without a time delay?

Reply:
For an asynchronous processing,it's common to use a waiting-time threshold to avoid hangup/block.

Since there are two threads,the main thread needs to wait cloudpoints transform thread finish its job. We can't assume it is called at the timing of "transforming just finish" every time. The worst case is: when wait_for is called, "cloudpoints transform thread" begin to transform cloudpoints. In such situation, the result is same as serial version. But if there are multi-cameras, it's still faster.

The hard code waiting-time threshold 30ms seems long. It won't wait such long time unless transform can't finish in 30ms which rarely happens.

Reply:
I think the delay is more likely to come from the transform not being available yet due to intermittent communication to the flight controller. But offloading something like that to another thread makes sense, so that the planner thread doesn't have `sleep`s in it.